compiler-plugin/src/main/java/io/ballerina/stdlib/persist/sql/compiler/codemodifier/QueryCodeModifierTask.java

```diff
@@ -0,0 +1,442 @@
+                        return null;
+                    }
+                    String fieldName = ((SimpleNameReferenceNode) fieldAccessNode.fieldName()).name().text();
+                    orderByClause.append(fieldName);
+                } else if (expression instanceof SimpleNameReferenceNode) {
+                    String fieldName = ((SimpleNameReferenceNode) expression).name().text();
+
+                    if (!(bindingPatternNode instanceof MappingBindingPatternNode)) {
+                        // If this is not mapping pattern there is compilation error
+                        return null;
+                    }
+                    boolean isCorrectField = false;
+                    SeparatedNodeList<BindingPatternNode> bindingPatternNodes =
+                            ((MappingBindingPatternNode) bindingPatternNode).fieldBindingPatterns();
+                    for (BindingPatternNode patternNode : bindingPatternNodes) {
+                        String field = ((FieldBindingPatternVarnameNode) patternNode).variableName().name().text();
+                        if (fieldName.equals(field)) {
+                            isCorrectField = true;
+                        }
+                    }
+                    if (!isCorrectField) {
+                        return null;
+                    }
+                    orderByClause.append(fieldName);
+                } else if (expression instanceof FunctionCallExpressionNode) {
+                    FunctionCallExpressionNode functionCallExpressionNode = (FunctionCallExpressionNode) expression;
+                    NameReferenceNode fun = functionCallExpressionNode.functionName();
+                    SeparatedNodeList<FunctionArgumentNode> arguments = functionCallExpressionNode.arguments();
+                    String referencedName =  Constants.INTERPOLATION_START_TOKEN + fun.toSourceCode().trim() +
+                            functionCallExpressionNode.openParenToken().text() +  arguments.get(0).toSourceCode() +
+                            functionCallExpressionNode.closeParenToken().text() +  Constants.INTERPOLATION_END_TOKEN;
+                    orderByClause.append(referencedName);
+                } else {
+                    // Persistent client does not support order by using parameters
+                    return null;
+                }
+                if (orderKeyNodes.get(i).orderDirection().isPresent()) {
+                    Token orderDirection = orderKeyNodes.get(i).orderDirection().get();
+                    if (orderDirection.text().equals(Constants.ASCENDING)) {
+                        orderByClause.append(Constants.SPACE).append(Constants.SQLKeyWords.ORDER_BY_ASCENDING);
+                    } else {
+                        orderByClause.append(Constants.SPACE).append(Constants.SQLKeyWords.ORDER_BY_DECENDING);
+                    }
+                    // Any typos are recognised as order by direction missing
```

Comment:
I think the default `ORDER BY` direction in MySQL and MSSQL is ascending. In this behaviour we would default to descending right?

Reply:
Updated now.